core/src/main/java/org/elasticsearch/index/query/SpanNearQueryBuilder.java

```diff
@@ -19,71 +19,172 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanNearQuery;
+import org.apache.lucene.search.spans.SpanQuery;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
 
+/**
+ * Matches spans which are near one another. One can specify slop, the maximum number
+ * of intervening unmatched positions, as well as whether matches are required to be in-order.
+ * The span near query maps to Lucene {@link SpanNearQuery}.
+ */
 public class SpanNearQueryBuilder extends AbstractQueryBuilder<SpanNearQueryBuilder> implements SpanQueryBuilder<SpanNearQueryBuilder> {
 
     public static final String NAME = "span_near";
 
-    private ArrayList<SpanQueryBuilder> clauses = new ArrayList<>();
+    static boolean DEFAULT_IN_ORDER = true;
+    static boolean DEFAULT_COLLECT_PAYLOADS = true;
```

Comment:
you mean that our public getter for it has no javadocs either? :)  you can have a look at `NearSpansPayloadOrdered` in lucene but you won't find much more javadocs, we can simply state that the flag controls whether we collect payloads or not :)